components/ModalDelete.js

```diff
@@ -4,36 +4,26 @@ import styled from "styled-components";
 export default function RemedyDeleteButton({ onDelete }) {
   const [showModal, setShowModal] = useState(false);
 
-  const handleDeleteClick = () => {
-    setShowModal(true);
-  };
-  const handleCloseModal = () => {
-    setShowModal(false);
-  };
+  function toggleModal() {
+    setShowModal(function (prevShowModal) {
+      return !prevShowModal;
+    });
+  }
 
-  const handleConfirmDelete = () => {
+  function handleConfirmDelete() {
     setShowModal(false);
     onDelete();
-  };
+  }
 
   return (
     <>
-      <DeleteButton onClick={handleDeleteClick}>Delete Remedy</DeleteButton>
-
+      <DeleteButton onClick={() => toggleModal()}>Delete Remedy</DeleteButton>
```

Comment:
You don't need to explicitly use a callback function if you pass the function as a reference

```suggestion
      <DeleteButton onClick={toggleModal}>Delete Remedy</DeleteButton>
```